### DISCUSSION TITLE
Koriki Update Test Release (Pre 1.0.2) 20231217

### DISCUSSION THREAD

**Discussion by acmeplus** [rg35xx-cfw/Koriki#91] [Announcements]
This is a test update that includes some of the new functionality for the next version of koriki/Batocera Lite for the RG3XX. This is not supposed to be complete, and while this build is stable there me functionality missing from the release.

### Some changes:
* Included both simplemenu and EmulationStation launchers
* Changed completely the kernel and init mechanism to resemble the stock one:
  * Leds should now work consistently
  * Battery should report charge correctly
  * Battery charge icon is now displayed when the unit is connected to the power
  * This version should work with all RG35XX hardware revisions
* USB port should now work:
  * USB gamepads should work, although those can only be mapped in Simplemenu
  * USB memories also work
  * USB Wifi adapters now work. Currently only EDIMAX EW-711Un (rtl8192cu based) and TP-Link W725N (rtl8188us/eu based) work. You will need to use EmulationStation to configure the wireless network
* ADB can be disabled/enabled via Simplemenu menu

### How to change the default launcher?

Create a file ``[SHARE]/system/customlauncher``, with one of these entries in plain text:
* ``simplemenu``: launches the standard Koriki Simplemenu launcher
* ``emulationstation``: launches Batocera EmulationStation
* ``retroarch``: Regular retroarch (like Lakka).

Report findings, comments, and/or suggestions in the discussion page for this release

<hr /><em>This discussion was created from the release <a href='https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_rg35xx_pre_1.0.2'>Koriki Update Test Release (Pre 1.0.2) 20231217</a>.</em>

**Comment by benni092** [comment]
Headphone jack detection is working (jay) but the volume bug is there where you must press a button otherwise it blasts your ears away (goes to 100% volume)

USB otg is working (i connect a xbox360 usb controller and the buttons can be mapped).

But i cant launch any standalone emu (retroarch cores works)
And the biggest issue for me is that i cant turn off the device. I go to menu select shutdown then is hangs for about 5-10 secs and emulationstation comes again. Tested it with simplemenu and it acts the same.
Have a nice day!

**Reply by acmeplus** [reply]
Let me know what's a simple scenario to reproduce the volume at 100% with the headphones, I tried several scenarios and I couldn't reproduce that, but probably I'm doing something different.

For the power off, I need to update the simplemenu binary, but you can use Volume Up + Power button to shutdown.

**Reply by benni092** [reply]
i simply boot into emulationstation. Volume is much louder (cant tell if its 100%) as soon as you press a button it becomes "normal".

When i boot a n64 game as an example, it fails to launch (thats another problem) and kicks me back to emulationstation, then the volume is again "loud".

**Comment by kartim282** [comment]
The system starts up with emulationstation, regardless of what I put in the customlauncher file. Starting any game gives me an unknown error 256. Pressing the power button asks for a configuration of a controller / keyboard.
I also can't turn off the device. It seems to restart emulationstation.

**Reply by benni092** [reply]
Have you copied the customlauncher file to
[SHARE]/system/? The file is in the batocera patition.

**Reply by acmeplus** [reply]
No, that's the correct location, the partition is called SHARE, it's actually the last partition of the SDCARD. In that partition you will see a ``system`` folder where you need to put the ``customlauncher`` file.

**Reply by kartim282** [reply]
I have a 2nd SD Card for the roms and put it in the system directory there. That seems to have worked. It now opened with simplemenu.

**Reply by acmeplus** [reply]
Yeah, that's good, forgot to mention that if you use a second card that's where you need to put the customlauncher file.

**Reply by kartim282** [reply]
Can I check logs somewhere for the error 256, when starting roms with emulationstation? Or is an unknown error really unknown?

**Reply by acmeplus** [reply]
Error 256 is either because there is no ``system/logs`` folder or because the card is corrupted and needs to be checked. You can use DiskUtility in Mac and run FirstAid to check for errors.

**Reply by kartim282** [reply]
I created a logs directory on the system folder on my 2nd sdcard. Now I can start games through emulationstation without error 256. Thanks!

**Reply by benni092** [reply]
Hello there!
I think i figured out why mupen64plus wont start.
the logfile says that it cant find /usr/lib/libmupen64plus.so.2.
I loocked in this folder and the files name is
libmupen64plus.so.2.0.0 .
Can this be the issue? Btw mupen starts with simplemenu without any problems.
Oh and ppsspp and drastic wont start (tested with simplemenu and emulationstation) but cant find any logs.
![IMG_20231219_204153](https://github.com/rg35xx-cfw/Koriki/assets/122157977/a4e5b8e8-7170-466a-b144-0812c21010d2)
[es_launch_stdout.log](https://github.com/rg35xx-cfw/Koriki/files/13719718/es_launch_stdout.log)

**Reply by kartim282** [reply]
I deleted a comment by accident. This as a reaction to "Have you copied the customlauncher file to [SHARE]/system/? The file is in the batocera patition.":
I put it in /Volumes/BATOCERA, which is the MacOS mount point for the partition. I'm not really sure if this is what you mean with [SHARE], or if this is something literal.

❯ ls
2100mAh-gpu.dtb adb_uImage boot customlauncher FSCK0000.REC modules tools uImage
2600mAh-gpu.dtb batocera-boot.conf boot_logo.bmp.gz def_config.bin kernel.dtb ramdisk.img uEnv.txt
❯ cat customlauncher
simplemenu

**Reply by acmeplus** [reply]
It’s fixed in the next release

**Comment by kartim282** [comment]
I cannot start GBA or N64 games. It returns right to emulationstation (ES). Stderr log of ES is empty. The regular log shows, as an example:

gameinfoxml: /tmp/game.xml
p1devicepath: /dev/input/event1
p1guid: 19000000010000000100000000010000
p1index: 0
p1name: RG35XX Gamepad
p1nbaxes: 0
p1nbbuttons: 17
p1nbhats: 0
rom: /userdata/roms/gba/Ace Combat Advance (USA, Europe).zip
system: gba
systemname: Game Boy Advance
Emulator: , Default core for gba:

**Reply by denizonm** [reply]
I am also having trouble with GBA, DS and PSP. Havent tried n64

**Comment by acmeplus** [comment]
New update is out: https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_rg35xx_v1.0.2_update

Keep using this discussion page for comments. The main change is a fix for the simplemenu poweroff and shutdown menu that was not working.

**Comment by acmeplus** [comment]
> Hello there! I think i figured out why mupen64plus wont start. the logfile says that it cant find /usr/lib/libmupen64plus.so.2. I loocked in this folder and the files name is libmupen64plus.so.2.0.0 . Can this be the issue? Btw mupen starts with simplemenu without any problems. Oh and ppsspp and drastic wont start (tested with simplemenu and emulationstation) but cant find any logs. ![IMG_20231219_204153](https://private-user-images.githubusercontent.com/122157977/291698161-a4e5b8e8-7170-466a-b144-0812c21010d2.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxMDQ5MzksIm5iZiI6MTcwMzEwNDYzOSwicGF0aCI6Ii8xMjIxNTc5NzcvMjkxNjk4MTYxLWE0ZTViOGU4LTcxNzAtNDY2YS1iMTQ0LTA4MTJjMjEwMTBkMi5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyMFQyMDM3MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMWU1ZDczZmZmOTc4OGRiZjVlNTUyZmVjNDZmMDI4MmU3ZGY3NWI3MDk0MjUxODc0ODAxZDVlMTk5NzZmODI0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.qSS6TXHdHV1evtmVHwBTLU7L55VS0jkgTR-pmMV6PlA) [es_launch_stdout.log](https://github.com/rg35xx-cfw/Koriki/files/13719718/es_launch_stdout.log)



> Hello there!
> I think i figured out why mupen64plus wont start.
> the logfile says that it cant find /usr/lib/libmupen64plus.so.2.
> I loocked in this folder and the files name is
> libmupen64plus.so.2.0.0 .

Which system are you using to test that simplemenu or ES? I've just tested with mupen64_rice_sa and is working without issues.

**Reply by benni092** [reply]
Yes on simplemenu mupen-sa rice works.
On Emulationstation mupen fails to start. Tested it with all emulator settings.

**Reply by acmeplus** [reply]
Ok. Will check what’s going on when I have some time.

**Reply by acmeplus** [reply]
Was curious about it and then I realized the issue is an optimization I did to make sure the games launch much faster in EmulationStation. The optimization launches games in 1-2s instead of the regular 7-8s that the default mechanism uses, but it does not work for standalone emulators.

It can be solved by a es_systems_custom.cfg overlay, but will provide some hints when I have time.

**Comment by paladinperry** [comment]
Just seen the release. Thank you guys ! I have the miyoo mini and the RG35XX, I exclusively use Koriki as I feel it is the best firmware available, even with its quirks. that being said the rg35xx has been stored since the last 2 updates as I had too many issues. So I'm really glad to see this today. Thank you so much for all your hard work, I'm staying loyal over here and Merry Christmas !

**Comment by mad1exx** [comment]
Yesterday I installed the new version of Koriki for RG35XX 1.0.2 update over the old version Koriki 1.0.1 (copied with file replacement) and chose a different shell, emulationstation. Today I discovered that in this shell the ports folder from the flash card is not shown and the psp emulator - PPSSPP does not work (it definitely worked on the previous version of koriki 1.0.1, I changed the shell to simplemenu and it also does not run iso on it. I also noticed: when you adjust the screen brightness with the volume and menu keys, both the volume and brightness are adjusted at the same time. The brightness level is not remembered after a reboot, the volume level at the time of launching games is louder than it was previously set in the shell. In any case, I will most likely stay with the koriki, as I am already impressed with the work done. Please accept my thanks!

**Comment by Charlychaps** [comment]
Hello there, 

Please help me I did copy customlauncher in [SHARE]/system/customlauncher 
I'v changed emulstation to simplemenu but nothing boot :(
Do I have to change the content of the customlauncher in Batocera partition?
Where I'm wrong.?

**Comment by acmeplus** [comment]
If you have problems with customlauncher just delete the file and the system should boot without issues using ES. The next release will default to simplemenu for the RG35XX original and EmulationStation for the Plus.

**Comment by acmeplus** [comment]
I’m aware that the current 1.0.2 has issues with PPSSPP (standalone or core) and NDS (drastic). That’s already fixed in an incoming update that should also fix brightness setting being saved during reboots as well as audio volume not behaving consistently.

**Comment by Charlychaps** [comment]
> If you have problems with customlauncher just delete the file and the system should boot without issues using ES. The next release will default to simplemenu for the RG35XX original and EmulationStation for the Plus.

In fact I like a lot emulstation but it doesn't shutdown at all, that's why I'd like to boot Koriki...
But ok then, I'll wait for your superb job! Many thanks Acmeplus

**Reply by acmeplus** [reply]
Yeah. That’s another bug there. The shutdown scripts from batocera/ES are not working right now. However you can use the Volume Up + Power button combo to shutdown

**Comment by jeifury** [comment]
Hello there.
I really like the EmulationStation  on this update, it seems kid friendly.  But I'm having issue on the artwork and also N64 doesn't work. Is there any work around for this to work?

**Reply by acmeplus** [reply]
By artwork you mean the rom thumbnails? Batocera/ES expects them in media/images. If you have roms with images in Imgs folder, create a new media/images and move the images there. Note the rom name without extension should match the thumbnail name (with .png or .jpg extension.

Regarding N64 not starting, check which emulator or libretro core you are using. Mupen64Plus standalone emulator does not read roms in archives (.zip or .7z) so you need them in .n64 or .z64 format. Libretto mupen64 can use compressed roms though, but the performance is not as good as mupen64plus standalone.

**Reply by jeifury** [reply]
The thumbnails are showing up when I'm using simplemenu but not if i'm using emulationstation. It's by default located in media/images in roms folder. 

I've tried switching cores on n64 but still no luck. They're all in .z64 and .n64 formats btw.

**Reply by acmeplus** [reply]
I assume you mean something like n64/media/images/game.png. If that's the path it should work for both Simplemenu and EmulationStation.

About N64, only suggestion is to try another rom, or to start retroarch manually in simplemenu and load core/load content to see if the rom format is the issues. Also remember you may need proper bios in [SHARE]/bios folder

**Reply by BeerBOFH** [reply]
I had to enable "Search For Local Art" under the Frontend Developer Options and place the artwork in "n64/images/[gamename].png"  to get artwork to show up in ES.

**Comment by paladinperry** [comment]
Hi guys. Just a quick one. How do I get simplemenu to launch. I'm on Win11 so I only see Windows based partitions i.e NTFS/FAT32
ES really isn't working for me and I miss the look of the old menu. Could anyone help me please ?
Thank you

**Reply by kartim282** [reply]
You should see the share partition (i.e. you can’t see the batocera partition), which is technically the 2nd partition. That is enough, this one should be fat32, readable by Windows. On it you have a system directory. Create a text file named customlauncher without extension (so no .txt at the end) in the system directory if it doesn’t exist. Just add “simplemenu” as text without the quotes.

If you use a second sd card for the roms, which only has one partition, do it in the system directory on there. In all cases, if the file already exists, just edit it and replace what’s in it.

The file on the 2nd sd card overrules the one on the 1st sd card, should you use two sd cards. If you have a 2nd card, you have to do it there!

**Reply by paladinperry** [reply]
Cheers mate. I change my drive labels in AOEMI frequently as I have so many external back up 4tb dirves for ROMS and rarities. It will just be hiding a partition with no label I think.
That info is great. thank you so much. 
I am a 2nd SD card user, so I am going intop it then the 'system' folder and what is the file name to edit (or do I create ?) 
Thanks again

**Reply by kartim282** [reply]
The file is called customlauncher. No extension.  Content is simplemenu. Reboot. Profit!

**Reply by paladinperry** [reply]
Cheers my man ! I'm going to it now. much appreciated and Merry Christmas !

**Comment by Res777** [comment]
A great work in progress update! I'm fine with the simple menu from previous releases, which I customized as I prefer (except the Koriki settings menu, because I couldn't find the strings to translate....where are they?), I'm waiting for the next update to fix some bugs,.
I also took a look at emulation station which seems much more complete, but the language doesn't change, upon reboot it returns to English. Is this another bug or am I doing something wrong?

**Reply by acmeplus** [reply]
I remember some folks had issues with ES translation not being saved, I'll investigate. Simplemenu cannot be translated, you need to redo the graphics to translate, and the menus are hardcoded (not my design) and cannot be translated unless you change the code. The new simplermenu+ that I'm writing with @alpgarcia includes translation support from scratch, but it's not finished yet.

**Comment by leonkasovan** [comment]
I can't find the setting for enabling/disabling ADB in simplemenu.
Where is the menu? Or how to do it?

**Reply by acmeplus** [reply]
Sorry, the scripts were not included in this update release, will be included in the next full release. However here are the scripts and you extract them and add them to [SHARE]/roms/ports. If you add them to the later you can then enable/disable ADB from EmulationStation.



[koriki_scripts.zip](https://github.com/rg35xx-cfw/Koriki/files/13762389/koriki_scripts.zip)

**Reply by leonkasovan** [reply]
Massive thanks for the script.

**Comment by Charlychaps** [comment]
Hello there, 
Merry Christmas! 
I want to apologize for what I've not done....
I said that I can't boot up to koriki instead of batocera blabla, in fact I just made a complete new installation....and it's amazing, it's working like a charm!!!
So thank you again for your work acmeplus! 
This console I've never played with it because of the lack of good launchers!
I have noticed when I insert the sdcard on my pc, win10 detect always the card has issue blabla and need to be "repaired" is it normal? Winbullshit? :)
Many thanks anyways

**Comment by denizonm** [comment]
This is amazing!! I realized you can put the apps into a ports folder in roms to access the settings from ES. Also I haven't seen this discussed anywhere but i got native pico working.

This is a chrismas miracle.

**Reply by Charlychaps** [reply]
hello, 
Could you please share the way to do it?
Many thanks

**Reply by denizonm** [reply]
Right now I can only get this working in simplemenu.

okay, so you buy pico8 and download the raspberry pi build. extract that to /bios/pico on your share partition. (so the pico, pico_dyn etc will be like /bios/pico/pico_dyn etc.)

replace the /system/.simplemenu/launchers/fake08_libretro with the version I have attached. (unarchive it first ofc)
[fake08_libretro.zip](https://github.com/rg35xx-cfw/Koriki/files/13771408/fake08_libretro.zip)

This will replace the fake08 launcher with the native pico launcher. If you wanted to, you could put it as an additional emulator setting but this is the way to do it with the least steps. plus, fake08 just doesn't work for me anyway so it's not a big loss.

Splore only kind of works. it downloads the list so you can browse, but doesnt download the games themselves for some reason.

---
If you have any problems, lmk.

**Reply by Charlychaps** [reply]
many thanks I'll try it ;)

**Reply by denizonm** [reply]
I found a way to kinda do it on ES as well. you put this into a ports folder in your roms folder:

[pico8_splore.sh.zip](https://github.com/rg35xx-cfw/Koriki/files/13771685/pico8_splore.sh.zip)

and put your pico roms to /system/.lexaloffle/pico-8/carts

This way you can use the splore browser to play the downloaded ones.

when you want to leave, press start on any of the roms in the splore menu
go down to options -> shutdown pico8

**Reply by acmeplus** [reply]
Yes, the official pico8 is supported and should work from ES or simplemenu

**Reply by denizonm** [reply]
Is there a ~~less hacky~~ more official way of doing this than the solution I found?

Where does ES get core associations from?

**Reply by acmeplus** [reply]
Yes. You can create a custom es_systems_pico8.cfg and (re)define the core association there. 

See this: https://wiki.batocera.org/emulationstation:customize_systems

**Reply by denizonm** [reply]
Thank you.
I followed the instructions here:
https://wiki.batocera.org/systems:pico8

And it didnt work for me so I thought koriki was doing something different compared to base batocera.

Im sure the answer is in the main documentation you sent.

Thank you again for your work.

---
The splore problem seems to be because wget is not installed. Is there a way I can install it manually?

**Reply by acmeplus** [reply]
I’ll add wget to the next release. You can try yo use a compatible binary but it’s better to wait until it’s native. 

Koriki uses a different system to launch roms. There are launcher scripts in system/.simplemenu/launchers and that should be similar to the one you created just addling the $1 argument to pass the rom name. Then you need to modify system/.simplemenu/section_groups/consoles.ini (or any other section where pico8 is defined) and add your new launcher. Don’t use spaces between launchers. 

Once you have you will be able to select that launcher on core options.

**Reply by denizonm** [reply]
okay, thank you!

**Comment by acmeplus** [comment]
v1.0.3 - 20240102 is now out: 
https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_batocera_rg35xx_20240102
